File: app/src/main/java/se/gustavkarlsson/skylight/android/gui/activities/main/fragments/aurora_chance/ChancePresenter.java
package se.gustavkarlsson.skylight.android.gui.activities.main.fragments.aurora_chance;

import android.widget.TextView;

import se.gustavkarlsson.skylight.android.evaluation.Chance;
import se.gustavkarlsson.skylight.android.evaluation.ChanceEvaluator;
import se.gustavkarlsson.skylight.android.evaluation.ChanceLevel;
import se.gustavkarlsson.skylight.android.models.AuroraReport;

public class ChancePresenter {
	private final TextView chanceTextView;
	private final ChanceEvaluator<AuroraReport> evaluator;

	public ChancePresenter(TextView chanceTextView, ChanceEvaluator<AuroraReport> evaluator) {
		this.chanceTextView = chanceTextView;
		this.evaluator = evaluator;
	}

	void update(AuroraReport report) {
		Chance chance = evaluator.evaluate(report);
		ChanceLevel chanceLevel = ChanceLevel.Companion.fromChance(chance);
		chanceTextView.setText(chanceLevel.getResourceId());
	}
}

File: app/src/main/java/se/gustavkarlsson/skylight/android/gui/activities/main/fragments/aurora_chance/ChancePresenter.kt
package se.gustavkarlsson.skylight.android.gui.activities.main.fragments.aurora_chance

import android.widget.TextView
import se.gustavkarlsson.skylight.android.evaluation.ChanceEvaluator
import se.gustavkarlsson.skylight.android.evaluation.ChanceLevel
import se.gustavkarlsson.skylight.android.models.AuroraReport

class ChancePresenter(
		private val chanceTextView: TextView,
		private val evaluator: ChanceEvaluator<AuroraReport>
) {

    fun update(report: AuroraReport) {
        val chance = evaluator.evaluate(report)
        val chanceLevel = ChanceLevel.fromChance(chance)
        chanceTextView.setText(chanceLevel.resourceId)
    }
}
